power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_pf_solver.hpp

```diff
@@ -211,7 +211,18 @@ template <symmetry_tag sym> class NewtonRaphsonPFSolver : public IterativePFSolv
           perm_(y_bus.size()) {}
 
     // Initilize the unknown variable in polar form
-    void initialize_derived_solver(YBus<sym> const& /* y_bus */, MathOutput<sym> const& output) {
+    void initialize_derived_solver(YBus<sym> const& y_bus, PowerFlowInput<sym> const& input, MathOutput<sym>& output) {
+        ComplexTensorVector<sym> linear_mat_data(y_bus.nnz_lu());
+        SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>> linear_sparse_solver{
+            y_bus.shared_indptr_lu(), y_bus.shared_indices_lu(), y_bus.shared_diag_lu()};
+        typename SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>>::BlockPermArray linear_perm(
```

Comment:
please use a `using` declaration at the start of the body of this function for readability